### PR TITLE
feat: Enable doxygen version to be configured

### LINKED ISSUE #29: doxygen version is limited to the latest version available via `apt-get`
I needed a feature provided in `1.9.6` of doxygen. Using the `apt-get` method in the action means you get the latest version from the package manager, and not necessarily the latest version available.

### PR DESCRIPTION
**Pull request by baynezy**
This makes the version of doxygen a paramter and allows the installation of versions that might not be available via `apt-get`.

Closes #29

### CODE REVIEW

**Comment by DenverCoder1** [comment]
Maybe the version number could be made an input variable so the user can choose the version and it could default to a specific version. What do you think?

**Comment by baynezy** [comment]
@DenverCoder1 - I did think about that, I can modify the PR. give me a second.

**Comment by baynezy** [comment]
@DenverCoder1 - don't merge this. Making this a parameter is more fiddly. Leave it with me.

**Comment by baynezy** [comment]
@DenverCoder1 - so I have made the changes to make this a parameter.

Note - in my testing, I discovered that this approach only works for versions 1.9.3 and higher. Previously this action was defaulting to 1.9.1. So this might be considered a breaking change. That is your call though.

**Comment by DenverCoder1** [comment]
Do you know there are significant differences between 1.9.1 and 1.9.6? If for most people, it shouldn't make a difference, it should be fine to make this a breaking change.

**Comment by baynezy** [comment]
There are no breaking changes according to their [changelog](https://www.doxygen.nl/manual/changelog.html#log_1_9_2). My code works with 1.9.1-1.12.0. I just need a capability they added in 1.9.6.

**Comment by baynezy** [comment]
@DenverCoder1 - is there anything more you want me to do on this?

**Comment by DenverCoder1** [comment]
I will test and this should be good :+1:

**Review by DenverCoder1 — APPROVED** [review]
Looks great! Thanks for the contribution :tada:

**Pull request merged**
